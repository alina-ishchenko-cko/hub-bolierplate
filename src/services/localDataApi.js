// @flow
// Set the localStorage keys namespace
const storageKeys = {
  loginStatus: 'cko.login',
  account: 'cko.account',
  dateRange: 'cko.account.dateRange',
  user: 'cko.user',
  userSession: 'cko.user.session',
};

// Local data api - stores and gets data
const localDataAPI = {
  storage: window.localStorage,
  /**
   * Gets the specified item
   * @param {string} key - item key
   * @returns {object}
   */
  getData(key: string): Object {
    const localData = this.storage.getItem(key);
    return JSON.parse(localData || '{}');
  },

  /**
   * Store the specified item
   * @param {string} key
   * @param {object} data
   */
  saveData(key: string, data: Object) {
    this.storage.setItem(key, JSON.stringify(data));
  },
};

/* -------------------------------------------------------------- EXPORTS */
/**
 * Login Session
 */
export const userSession = {
  get() {
    return localDataAPI.getData(storageKeys.userSession);
  },
  save() {
    localDataAPI.saveData(storageKeys.userSession, {
      lastDate: new Date().toISOString(),
    });
  },
};

/**
 * User
 */
export const user = {
  get() {
    return localDataAPI.getData(storageKeys.user);
  },
  save(data: Object) {
    localDataAPI.saveData(storageKeys.user, data);
  },
};

/**
 * Accounts
 */
export const accounts = {
  get() {
    return localDataAPI.getData(storageKeys.account);
  },
  save(data: Object) {
    localDataAPI.saveData(storageKeys.account, data);
  },
};

/**
 * Date
 */
export const dateRange = {
  get() {
    return localDataAPI.getData(storageKeys.dateRange);
  },
  save(data: Object) {
    localDataAPI.saveData(storageKeys.dateRange, data);
  },
};

/**
 * Get the user token
 * @returns {string}
 */
export function getToken(): string {
  const userData = user.get();
  return userData.data ? userData.data.token : '';
}

export const login = {
  get() {
    return localDataAPI.getData(storageKeys.loginStatus);
  },
  passed(data: Object) {
    localDataAPI.saveData(storageKeys.loginStatus, { data: true });
  },
};

/**
 * Checks if user is authenticated
 * @returns {boolean}
 */
export function isAuthenticated(): boolean {
  const currentUser = user.get().data;
  const loginStatus = login.get().data;
  return (
    !!(currentUser && currentUser.email !== '' && currentUser.token !== '') ||
    !!loginStatus
  );
}

/**
 * Clears all the data in the localStorage
 */
export function clearData() {
  localDataAPI.storage.clear();
}
