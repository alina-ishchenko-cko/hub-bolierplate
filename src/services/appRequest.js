// @flow
import axios from 'axios';
import pickBy from 'lodash/pickBy';
import identity from 'lodash/identity';
import isEmpty from 'lodash/isEmpty';
import * as localApi from 'services/localDataApi';

const appData: {
  url: string,
  token: string,
  currencyId: string,
} = {
  url: '',
  token: '',
  currencyId: '',
};

/**
 * Sets the current user's currencyId
 * @param {string} url
 */
export function setUrl(url: string): void {
  appData.url = url;
}

/**
 * Sets the current user's token
 * @param {string} token
 */
export function setToken(token: string): void {
  appData.token = token;
}

/**
 * Sets the current user's currencyId
 * @param {string} value - currency ID
 */
export function setCurrencyId(value: string): void {
  appData.currencyId = value;
}

/** Gets the current user's currencyId */
export function getCurrencyId(): string {
  if (appData.currencyId) {
    return appData.currencyId;
  }

  const user = localApi.user.get();
  // presumes user is guaranteed to have a data property
  return user && user.data ? user.data.displayCurrencyId : '';
}

/** Gets the API endpoint root domain */
export function getUrl(): string {
  return appData.url;
}

/** Gets the current user's token */
export function getToken(): string {
  const token = appData.token;
  return token;
}

/**
 * Maps RESTful actions.
 * All actions accepts the following arugments
 * (path = String, data = Object, reqHeaders = Object)
 * @returns {object}
 * - get()
 * - post()
 * - put()
 * - delete()
 */

export const ajxPromise = {
  get(path: string = '', data: Object = {}, reqHeaders: Object = {}) {
    return handleRequest('get', path, data, reqHeaders);
  },
  post(path: string = '', data: Object = {}, reqHeaders: Object = {}) {
    return handleRequest('post', path, data, reqHeaders);
  },
  put(path: string = '', data: Object = {}, reqHeaders: Object = {}) {
    return handleRequest('put', path, data, reqHeaders);
  },
  delete(path: string = '', data: Object = {}, reqHeaders: Object = {}) {
    return handleRequest('delete', path, data, reqHeaders);
  },
};

/** Clears the stored currencyId and token */
export function clearAppData(): void {
  appData.currencyId = '';
  appData.token = '';
}

/**
 * Makes Ajax request and returns a promise
 * @param {string} method
 * @param {string} path
 * @param {object} data
 * @param {object} reqHeaders
 * @returns {promise}
 */
function handleRequest(
  method: string = '',
  path: string = '',
  data: ?Object = {},
  reqHeaders: ?Object = {}
): Promise<Object> {
  let token = getToken();
  let endpoint = getUrl();
  let headers = { ...reqHeaders };
  if (isEmpty(data)) data = null;
  if (token) headers['X-AuthToken'] = token;
  if (isEmpty(headers)) headers = null;

  // Remove Object undefinded or null props
  // before making the request.
  const reqObj = pickBy(
    {
      url: `${endpoint}/hub${path}`,
      method,
      data,
      headers,
    },
    identity
  );

  // Make request
  return axios(reqObj);
}
