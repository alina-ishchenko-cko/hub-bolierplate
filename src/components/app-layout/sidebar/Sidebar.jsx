// @flow
import * as React from 'react';
import isEqual from 'lodash/isEqual';
import CkoIcon from 'components/ui/icon';
import classNames from 'classnames';
import { getMenuArray } from 'utils/ui.util';
import * as localApi from 'services/localDataApi';
import { FlexItem, FlexColumn } from 'components/ui/flex/';
import { AppSidebarStyled } from './Sidebar.sc';
import * as typed from 'store/reducers/flow-type';
import AccountSelector from '../account-selector/';
import MainMenu from '../main-menu/';
import AppFooter from '../footer/';
import Logo from '../logo/';
import { auth } from 'services/security/authorisation';

type Props = {
  currentUser: typed.LoginData,
  global: typed.Account,
  selected: typed.Selected,
  history: {
    push: Function,
  },
  location: {
    pathname: string,
  },
  getAccountAssets: Function,
  setSelection: Function,
  getAccounts: Function,
  globalLookUp: Function,
  logout: Function,
  toggleSideBar: Function,
};

type State = {
  disableMenu: boolean,
  openSidebar: number,
};

export default class Sidebar extends React.Component<Props, State> {
  static defaultProps = {
    history: {},
    location: {},
    global: {},
    currentUser: {},
  };

  state = {
    disableMenu: false,
    openSidebar: 0,
  };

  componentDidMount() {
    this.setInitData();
  }

  shouldComponentUpdate(nextProps: Props, nextState: State) {
    return !isEqual(this.props, nextProps) || !isEqual(this.state, nextState);
  }

  componentDidUpdate(prevProps: Props) {
    const { accounts, businesses } = this.props.global;
    const { permissions } = this.props.currentUser;

    if (permissions && permissions.length > 0) {
      this.checkViewPermission();
    }

    // Check if data has been updated and only single accounts
    // Get the selected account assets
    if (this.isSingleAccount(accounts)) {
      const accBusiness = businesses[accounts[0].accountId];
      if (
        this.hasDataChanged(prevProps) &&
        !this.props.global.assetsLoading &&
        !accBusiness
      ) {
        this.props.getAccountAssets(accounts[0].accountId);
        return;
      }

      // Check account has businesses
      if (
        accBusiness &&
        accBusiness.length > 0 &&
        !this.props.selected.account.id
      ) {
        this.initSelector(accounts[0], accBusiness);
      }
    }
  }

  /** Sets the inital data for date range and global configs */
  setInitData() {
    // Check if data is cached
    const accountsData = localApi.accounts.get();

    if (accountsData.data === void 0) {
      this.props.getAccounts();
      this.props.globalLookUp();
    }
  }

  initSelector(account: Object, businesses: Array<Object>) {
    const params = {
      ...this.props.selected,
      account: {
        id: parseInt(account.accountId, 10),
        title: account.name,
      },
    };

    if (businesses.length === 1) {
      const business = businesses[0];
      const isSingleChannel = !!(business.channels.length === 1);
      params.business = {
        id: business.propertyId,
        title: business.propertyName,
      };
      params.channel = {
        id: isSingleChannel ? business.channels[0].channelId : null,
        title: isSingleChannel ? business.channels[0].channelName : '',
      };
    }

    // Pass selection to setSelection() action
    this.props.setSelection(params);
  }

  checkViewPermission() {
    const currentPage = this.props.location.pathname;
    const hasNoTransactionsAccess =
      currentPage.includes('/transactions') && !auth.canView('transactions');
    const hasNoStatementsAccess =
      currentPage.includes('/statements') && !auth.canView('Deposits::Viewing');

    if (hasNoTransactionsAccess || hasNoStatementsAccess) {
      this.props.logout();
    }
  }

  /**
   * Checks if NOT multiple accounts
   * @returns {boolean}
   */
  isSingleAccount(accounts: Array<Object>): boolean {
    return !!(accounts.length === 1);
  }

  isSingleBusiness(businesses: Array<Object>): boolean {
    return !!(businesses.length === 1);
  }

  /**
   * Checks if the state or props has changed
   * @param {Object} nextProps
   * @returns {boolean}
   */
  hasDataChanged(prevProps: Props): boolean {
    return !isEqual(prevProps.global, this.props.global);
  }

  /** Handles the logout
   * - Clears the store state
   * - redirects to the login page
   * - refreshes the page
   */
  handleLogout = (e: SyntheticEvent<HTMLLIElement>) => {
    e.preventDefault();
    e.stopPropagation();
    // Clear the store states
    this.props.logout();

    // Redirect to login page
    this.props.history.push('/login');

    // Refresh the whole page
    window.location.reload();
  };

  /** Get the selected account|business|channel */
  getSelected() {
    return this.props.selected;
  }

  /** Handles the account change */
  onAccountChange = (selected: Object) => {
    const businesses = this.props.global.businesses[selected.account.id];
    // Check if accountId Set and
    // The accounts asset does not exist
    if (selected.account.id && !businesses) {
      this.props.getAccountAssets(selected.account.id);
    }

    this.props.setSelection(selected);
  };

  handleSelectorOpen = (disableMenu: boolean) => {
    this.setState({ disableMenu });
  };

  expandSidebar = () => {
    this.setState({ openSidebar: 2 });
    this.handleSidebarToggle(2);
  };

  collapseSidebar = () => {
    this.setState({ openSidebar: 1 });
    this.handleSidebarToggle(1);
  };

  handleSidebarToggle(openSidebar: number) {
    this.props.toggleSideBar(openSidebar);
  }

  render() {
    const { global, currentUser } = this.props;
    const className = {
      openBtn: classNames({
        'collapse-btn': true,
        'open-trigger': true,
        show: this.state.openSidebar === 1,
        hide: this.state.openSidebar === 2,
      }),
      closeBtn: classNames({
        'collapse-btn': true,
        'close-trigger': true,
        show: this.state.openSidebar === 2,
        hide: this.state.openSidebar === 1,
      }),
    };
    return (
      <AppSidebarStyled id="sidebar" className="sidebar-menu">
        <FlexColumn className="inner-sider">
          <FlexItem grow="0">
            <Logo />
          </FlexItem>
          <FlexItem grow="0">
            <AccountSelector
              expandSidebar={this.expandSidebar}
              assetsLoading={global.assetsLoading}
              selected={this.getSelected()}
              onChange={this.onAccountChange}
              loading={global.loading}
              businesses={global.businesses}
              accounts={global.accounts}
              onOpen={this.handleSelectorOpen}
              isGodUser={currentUser.isGodUser || currentUser.isSuperAdmin}
            />
          </FlexItem>
          <FlexItem grow="0">
            <MainMenu
              disable={this.state.disableMenu}
              currentPath={this.props.location.pathname}
              onLogout={this.handleLogout}
              menuData={getMenuArray()}
              username={currentUser.name}
            />
            <button className={className.openBtn} onClick={this.expandSidebar}>
              <CkoIcon name="right" />
            </button>
            <button
              className={className.closeBtn}
              onClick={this.collapseSidebar}>
              <CkoIcon name="left" />
            </button>
          </FlexItem>
          <FlexColumn justify="flex-end">
            <FlexItem grow="0">
              <AppFooter />
            </FlexItem>
          </FlexColumn>
        </FlexColumn>
      </AppSidebarStyled>
    );
  }
}
