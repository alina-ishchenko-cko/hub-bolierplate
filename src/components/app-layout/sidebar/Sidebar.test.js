import React from 'react';
import MainMenu from '../main-menu/';
import Sidebar from './Sidebar';
import { MemoryRouter } from 'react-router-dom';
import * as localApi from 'services/localDataApi';
import AppLogo from '../logo/';
import AppFooter from '../footer/';
import AccountSelector from '../account-selector/';
import { AppSidebarStyled } from './Sidebar.sc';

describe('Sidebar', () => {
  let AppComp, mockProps;

  beforeEach(() => {
    mockProps = {
      currentUser: {
        permissions: [],
      },
      global: {
        assetsLoading: false,
        accounts: [
          {
            accountId: 1000,
          },
        ],
        businesses: {},
      },
      selected: {
        account: {
          id: null,
          title: 'account title',
        },
        business: {
          id: null,
          title: '',
        },
        channel: {
          id: null,
          title: '',
        },
      },
      history: {
        push: jest.fn(),
      },
      location: {
        pathname: '/dashboard',
      },
      getAccountAssets: jest.fn(),
      setSelection: jest.fn(),
      getAccounts: jest.fn(),
      globalLookUp: jest.fn(),
      logout: jest.fn(),
      toggleSideBar: jest.fn(),
    };
    AppComp = shallow(<Sidebar {...mockProps} />);
  });

  afterEach(() => {
    AppComp = null;
    localApi.clearData();
  });

  it('renders without crashing', () => {
    expect(AppComp.exists()).toBe(true);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Sidebar {...mockProps} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain required styled components', () => {
    expect(AppComp.find(AppSidebarStyled)).toHaveLength(1);
  });

  it('should contain required components', () => {
    expect(AppComp.find(AccountSelector)).toHaveLength(1);
    expect(AppComp.find(MainMenu)).toHaveLength(1);
    expect(AppComp.find(AppLogo)).toHaveLength(1);
    expect(AppComp.find(AppFooter)).toHaveLength(1);
  });

  it('should setup the data on mount', () => {
    mockProps.globalLookUp.mockClear();
    mockProps.getAccounts.mockClear();
    mockProps.currentUser = { token: '234567890' };
    AppComp = shallow(<Sidebar {...mockProps} />);
    expect(mockProps.getAccounts).toHaveBeenCalledTimes(1);
    expect(mockProps.globalLookUp).toHaveBeenCalledTimes(1);
  });

  it('should load assets and set selection on refresh', () => {
    mockProps.globalLookUp.mockClear();
    mockProps.getAccounts.mockClear();
    mockProps.getAccountAssets.mockClear();
    mockProps.setSelection.mockClear();

    const selection = {
      account: { id: 1000, title: 'account title' },
      business: { id: 1003, title: 'business name' },
      channel: { id: 10004, title: 'channel title' },
    };
    mockProps.global.accounts = [];
    mockProps.currentUser = { token: '234567890' };
    AppComp = shallow(<Sidebar {...mockProps} />);
    expect(mockProps.getAccounts).toHaveBeenCalledTimes(1);
    expect(mockProps.globalLookUp).toHaveBeenCalledTimes(1);

    // Create prop change when account response
    AppComp.setProps({
      global: {
        assetsLoading: false,
        accounts: [
          {
            accountId: 1000,
          },
        ],
        businesses: {},
      },
    });

    expect(mockProps.getAccountAssets).toHaveBeenCalledTimes(1);

    // Create prop for business asset resposnse
    AppComp.setProps({
      global: {
        assetsLoading: false,
        accounts: [
          {
            accountId: 1000,
            name: 'account title',
          },
        ],
        businesses: {
          '1000': [
            {
              propertyId: 1003,
              propertyName: 'business name',
              channels: [
                {
                  channelId: 10004,
                  channelName: 'channel title',
                },
              ],
            },
          ],
        },
      },
    });

    expect(mockProps.setSelection).toHaveBeenCalledTimes(1);
    expect(mockProps.getAccountAssets).toHaveBeenCalledWith(
      selection.account.id
    );
    expect(mockProps.setSelection).toHaveBeenCalledWith(selection);
  });

  it('should logout user', () => {
    const tempMenu = AppComp.find(MainMenu).prop('onLogout');
    tempMenu({
      preventDefault: () => {},
      stopPropagation: () => {},
    });
    //
    expect(mockProps.logout).toHaveBeenCalledTimes(1);
    expect(mockProps.history.push).toHaveBeenCalledTimes(1);
    expect(mockProps.history.push).toHaveBeenCalledWith('/login');
  });

  it('should expand sidebar', () => {
    AppComp.setState({
      openSidebar: 1,
    });
    const Btn = AppComp.find('button').at(0);
    Btn.simulate('click');
    expect(AppComp.state('openSidebar')).toBe(2);
    expect(mockProps.toggleSideBar).toHaveBeenCalledWith(2);
    expect(mockProps.toggleSideBar).toHaveBeenCalledTimes(1);
  });

  it('should collapse sidebar', () => {
    AppComp.setState({
      openSidebar: 2,
    });
    const Btn = AppComp.find('button').at(1);
    Btn.simulate('click');
    expect(AppComp.state('openSidebar')).toBe(1);
    expect(mockProps.toggleSideBar).toHaveBeenCalledWith(1);
    expect(mockProps.toggleSideBar).toHaveBeenCalledTimes(1);
  });
});
