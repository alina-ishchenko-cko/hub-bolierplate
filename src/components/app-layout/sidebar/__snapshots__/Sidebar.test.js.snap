// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
.css-0 {
  background: transparent;
  height: 100vh;
  position: fixed;
  z-index: 9;
}

.css-0 .inner-sider {
  width: 100%;
  height: 100%;
  position: relative;
}

.css-0 .collapse-btn {
  border: none;
  width: 90%;
  display: none;
  margin: 30px auto;
  border-radius: 3px;
  line-height: 30px;
  background: rgba(186,192,206,0.25);
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  line-height: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: auto;
}

.css-2 {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin: 0;
  padding: 0;
}

.css-3 {
  text-align: center;
  width: 100%;
  margin-top: 0.4375rem;
  margin-bottom: 1.25rem;
}

.css-3 .cko-icon {
  width: 35px;
}

.css-4 {
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-transform: none;
  line-height: 1px;
  width: auto;
  border-radius: 3px;
}

.css-4.placeholder {
  background-color: #dfe1ea;
}

.css-4 svg {
  width: auto !important;
  height: auto !important;
  max-width: 100%;
}

.css-4 .white {
  fill: #fff;
}

.css-4 .dark {
  fill: #8b94a9;
}

.css-4 .light {
  fill: #bac0ce;
}

.css-4 .dark,
.css-4 .light,
.css-4 .dark,
.css-4 .grey {
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.css-4 .icon-sidebar-transactions .dark,
.css-4 .icon-sidebar-dashboard .dark {
  fill: #8b94a9;
}

.css-4 .icon-sidebar-transactions .light,
.css-4 .icon-sidebar-dashboard .light {
  fill: #bac0ce;
}

.css-4 .icon-sidebar-plans .grey {
  fill: #fff;
}

.css-4 .icon-calendar * {
  fill: #ACB2BF;
}

.css-4 .icon-right-arrow *,
.css-4 .icon-left-arrow *,
.css-4 .icon-down-arrow * {
  fill: #abb3c4;
}

.css-4 .icon-paypal-full * {
  fill: #2464b4;
}

.css-6 {
  cursor: pointer;
  position: relative;
  width: 100%;
  height: auto;
  margin-bottom: 0.8461538461538461em;
  z-index: 8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.css-6.list-active .selected-business {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.css-6.list-active .channel-wrap {
  display: block;
}

.css-6 .selected-business i {
  display: block;
}

.css-8 {
  width: 100%;
  overflow: hidden;
  font-weight: 500;
  background-color: #fff;
  position: relative;
  z-index: 1;
  border-radius: 3px;
  box-shadow: 0px 2px 5px 0px rgba(121,131,149,0.2);
}

.css-8 .parent-title {
  display: block;
  text-transform: uppercase;
  color: #ADB0B8;
  font-size: 0.7692307692307693em;
  padding: 0;
  margin: 0;
  width: 86%;
  text-overflow: ellipsis;
  overflow: hidden;
  height: 15px;
  white-space: nowrap;
}

.css-8 .child-title {
  color: #1E242E;
  font-size: 1.2307692307692308em;
  width: 86%;
  text-overflow: ellipsis;
  overflow: hidden;
}

.css-8 i {
  position: absolute;
  font-style: normal;
  top: 25px;
  right: 20px;
  color: #798395;
}

.css-8 i:before {
  font-family: Flaticon;
  content: '\\f104';
  position: absolute;
  top: 0;
  right: 0;
}

.css-9 {
  width: 100%;
  margin: 5% auto;
}

.css-9 .dots {
  width: 100%;
}

.css-9 .dots > div {
  background-color: #ADB0B8;
  border-radius: 100%;
  width: 12px;
  height: 12px;
  margin: 2px;
  display: inline-block;
  -webkit-animation: animation-mu3yif 0.7s 0s infinite linear;
  -webkit-animation: animation-mu3yif 0.7s 0s infinite linear;
  animation: animation-mu3yif 0.7s 0s infinite linear;
}

.css-9 .dots > div:nth-child(2n-1) {
  -webkit-animation-delay: -0.35s;
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

.css-10 {
  border: none;
  background: transparent;
  position: relative;
}

.css-10 .overlay {
  position: absolute;
  background: transparent;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.css-10 .menu-list {
  margin: 0;
  padding: 0;
}

.css-10 .menu-list.disable {
  opacity: 0.3;
}

.css-10 .menu-item {
  margin: 0;
  height: 46px;
  line-height: 46px;
  color: #798395;
}

.css-10 .menu-item .cko-icon {
  max-width: 18px;
  max-height: 16px;
}

.css-10 .menu-item a {
  font-weight: 400;
  color: #798395;
  display: inline-block;
}

.css-10 .menu-item.active-menu a,
.css-10 .menu-item a:hover,
.css-10 .menu-item:hover {
  color: #3A6FD2;
}

.css-10 .menu-item.active-menu .cko-icon svg .dark {
  fill: #5b91e3;
}

.css-10 .menu-item.active-menu .cko-icon svg .light {
  fill: #a1c6ff;
}

.css-10 .menu-item:hover .cko-icon svg .dark {
  fill: #464f60;
}

.css-10 .menu-item:hover .cko-icon svg .light {
  fill: #b2b8c6;
}

.css-10 .menu-item span {
  vertical-align: middle;
  font-size: 1.0769230769230769em;
}

.css-10 .menu-item .cko-icon {
  font-size: 1.2692307692307692em;
}

.css-10 .logout {
  cursor: pointer;
}

.css-10 .side-info {
  margin-top: 1.25rem;
  margin-bottom: 0.625rem;
  padding: 0 0 0 1.25rem;
  font-weight: 500;
  text-overflow: ellipsis;
}

.css-10 .side-info p {
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.css-10 .side-info .title {
  font-size: 0.7692307692307693em;
  text-transform: uppercase;
  color: #ADB0B8;
}

.css-10 .side-info .username {
  font-size: 1.0769230769230769em;
  color: #464F60;
}

.css-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  line-height: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: auto;
}

.css-12 {
  padding: 0;
  margin-top: 30px;
  width: 100%;
}

.css-13 {
  margin: 0;
  font-weight: 400;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  border-radius: 3px;
  text-align: center;
  border: solid 1px transparent;
  padding: 0 1.25rem;
  font-size: 1em;
  line-height: 42px;
  width: auto;
  cursor: pointer;
  color: #fff;
  background: linear-gradient(to top,#4179e4,#4780ec);
  box-shadow: 0 1px 3px 0 rgba(58,128,232,0.45);
}

.css-13 .cko-icon svg * {
  fill: #fff;
}

.css-13:hover {
  background: linear-gradient(to top,#5993ff,#558ffa);
  box-shadow: 0 2px 5px 0 rgba(58,128,232,0.45);
}

.css-13:active,
.css-13:focus {
  background: linear-gradient(to bottom,#4179e4,#4780ec);
  box-shadow: inset 0 2px 5px 0 rgba(44,91,179,0.45);
}

.css-13 .cko-icon {
  margin-right: 15px;
}

.css-17 {
  display: inline-block;
  vertical-align: middle;
  margin-left: 1.25rem;
  padding-bottom: 2px;
  margin-bottom: 1px;
}

.css-17 a {
  font-size: 0.9230769230769231em;
  display: block;
  color: #ADB0B8;
  margin: 0;
  padding: 0;
}

@media only screen and (max-width:1700px) {
  .css-0 .open-trigger {
    display: block;
  }
}

@media only screen and (min-width:1700px) {
  .css-0 .close-trigger {
    display: block;
  }
}

<aside
  className="sidebar-menu css-0"
  id="sidebar"
>
  <div
    className="cko-flex-column inner-sider css-1"
  >
    <div
      className="cko-flex-item  css-2"
    >
      <div
        className="menu-logo css-3"
      >
        <span
          className="cko-icon  css-4"
        >
          <span
            src="/assets/svgs/logo.svg"
          />
        </span>
      </div>
    </div>
    <div
      className="cko-flex-item  css-2"
    >
      <div
        className="account-selector"
      >
        <div
          className="css-6"
          id="accountSelect"
        >
          <div
            className="selected-business css-8"
            onClick={[Function]}
          >
            <div
              className="cko-app-loading text-center css-9"
              size={undefined}
            >
              <div
                className="dots"
              >
                <div />
                <div />
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cko-flex-item  css-2"
    >
      <div
        className="css-10"
      >
        <ul
          className="menu-list"
        >
          <li
            className="side-info"
          >
            <p
              className="title"
            >
              Account configurations
            </p>
            <p
              className="username"
            >
              
            </p>
          </li>
          <li
            className="menu-item"
          >
            <a
              href="/settings"
              id="settings-link"
              onClick={[Function]}
            >
              <span
                className="cko-icon  css-4"
              >
                <span
                  src="/assets/svgs/sidebar-settings.svg"
                />
              </span>
              <span
                className="menu-title"
              >
                Settings
              </span>
            </a>
          </li>
          <li
            className="menu-item logout"
          >
            <a
              href="/login"
              id="logout-link"
              onClick={[Function]}
            >
              <span
                className="cko-icon  css-4"
              >
                <span
                  src="/assets/svgs/sidebar-logout.svg"
                />
              </span>
              <span
                className="menu-title"
              >
                Log out
              </span>
            </a>
          </li>
        </ul>
      </div>
      <button
        className="collapse-btn open-trigger"
        onClick={[Function]}
      >
        <span
          className="cko-icon  css-4"
        >
          <span
            src="/assets/svgs/right.svg"
          />
        </span>
      </button>
      <button
        className="collapse-btn close-trigger"
        onClick={[Function]}
      >
        <span
          className="cko-icon  css-4"
        >
          <span
            src="/assets/svgs/left.svg"
          />
        </span>
      </button>
    </div>
    <div
      className="cko-flex-column  css-11"
    >
      <div
        className="cko-flex-item  css-2"
      >
        <div
          className="menu-footer css-12"
        >
          <button
            className="cko-button  css-13"
            onClick={[Function]}
            size="large"
            type="primary"
            value="Docs"
          >
            <span
              className="value"
            >
              Docs
            </span>
          </button>
          <div
            className="footer-about css-17"
          >
            <a
              target="_blank"
            >
              About Checkout
            </a>
            <a
              target="_blank"
            >
              Privacy Policy
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;
